File: packages/sdk-java/src/main/java/xyz/alphahuman/sdk/AlphahumanError.java
package xyz.alphahuman.sdk;

/**
 * Thrown when the Alphahuman API returns a non-2xx response or a non-JSON body.
 */
public class AlphahumanError extends RuntimeException {

    private final int status;
    private final Object body;

    public AlphahumanError(String message, int status, Object body) {
        super(message);
        this.status = status;
        this.body = body;
    }

    public AlphahumanError(String message, int status) {
        this(message, status, null);
    }

    public int getStatus() {
        return status;
    }

    public Object getBody() {
        return body;
    }
}

File: packages/sdk-csharp/src/Alphahuman.Sdk/AlphahumanError.cs
namespace Alphahuman.Sdk;

public class AlphahumanError : Exception
{
    public int Status { get; }
    public string Body { get; }

    public AlphahumanError(string message, int status, string body)
        : base(message)
    {
        Status = status;
        Body = body;
    }
}
